src/Http/Place/LegacyPlaceRequestBodyParser.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Http\Place;
+
+use CultuurNet\UDB3\Http\Request\Body\RequestBodyParser;
+use Psr\Http\Message\ServerRequestInterface;
+use stdClass;
+
+final class LegacyPlaceRequestBodyParser implements RequestBodyParser
+{
+    public function parse(ServerRequestInterface $request): ServerRequestInterface
+    {
+        $data = $request->getParsedBody();
+
+        if (!($data instanceof stdClass)) {
+            return $request;
+        }
+        $data = clone $data;
+
+        $mainLanguage = $data->mainLanguage ?? null;
+
+        if ($mainLanguage && isset($data->name) && is_string($data->name)) {
+            $data->name = (object) [
+                $data->mainLanguage => $data->name,
+            ];
+        }
+
+        if (isset($data->type) && $data->type instanceof stdClass) {
+            $data->terms = [
+                (object) [
+                    'id' => $data->type->id,
+                    'label' => $data->type->id,
+                    'domain' => $data->type->id,
```

Comment:
1. I think you accidentally copy-pasted the `->id` for all three here
2. It's possible that these properties are not set in the request body, so I'd only set them if they are available